internal/cloudapi/v2/openapi.v2.yml

```diff
@@ -598,10 +551,17 @@ components:
           example: ['123456789012']
           items:
             type: string
-    GCPUploadRequestOptions:
+    AWSS3UploadOptions:
+      type: object
+      required:
+        - region
+      properties:
+        region:
+          type: string
+          example: 'eu-west-1'
+    GCPUploadOptions:
       type: object
       required:
-        - bucket
```

Comment:
@thozza Thoughts on dropping this and fixing/pinning the bucket in composer itself? We'll do the same for aws, just wanted to double-check that we can do this for GCP as well.

Reply:
What is the plan WRT sharing the built image with the customer in the nearest region? GCP has ~3 major geographic multi-regions, so we may need to decide which one to use for uploading the image and then importing it. I'm not sure how will composer determine the desired location. If there will be a way to specify it to composer, then yes, the logic to choose the appropriate storage bucket can be implemented in composer and based on its configuration. I just expect what we don't necessarily want to always default to North America region, although it may be good enough to start with.

Reply:
Ah, so I see that the `region` is still part of the `GCPUploadOptions`. In this case we will need to implement the logic to choose the appropriate bucket in composer based on the region...

Reply:
We could do something like "bucket + '.' + region" ; so `image-builder.service.eu` or whatever.

Reply:
How will this problem solved for AWS? Does it exist there at all? I'm not completely sure if choosing a bucket in the appropriate region matters too much for AWS.

Combining `bucket` and `region` into single string seems unnecessarily complicated to me. It definitely does not make sense to expose the `bucket` in `image-builder` API. As for the composer Cloud API, I would propose two options:

1. We keep the `bucket` in the request, instead of combining `bucket` and `region` into single string.
2. Remove the `bucket` from the request and implement a logic and configuration in composer, which will choose the appropriate bucket based on the requested region and have the code be shared by all cloud providers where this matters (my expectation is that it could be AWS and GCP at least?).
3. ... (open to suggestions)

Reply:
Ah it seems aws actually has the same restriction as gcp here actually, buckets need to be uniquely named, cross-region.

I don't think we should allow users to specify it here either, since we're the only ones who control these buckets, and if at one point a 2nd service other than our own depends on the cloudapi they can't really choose these anyway, we'd have to tell them what to use.